### DISCUSSION TITLE
Create sig-simulation and migrate some responsibilities from sig-core and sig-content

### DISCUSSION THREAD

**Discussion by greerdv** [o3de/o3de#8205] [General]
There is a proposal to create a new sig-simulation, which would have responsibilities including:
- physics, cloth, destruction (currently in sig-core)
- AI (e.g. pathfinding) (currently in sig-core?)
- animation (currently in sig-content)

The reasons motivating this proposal:
- The main reason is that sig-core and sig-content are kind of miscellaneous grab bags of responsibility currently. Sig-core has a roughly 50-50 split between physics and systems issues, which are quite disjoint areas. There is not much overlap in interest between the two areas, which means that about half of the time spent in e.g. triage meetings is not highly relevant to attendees. It also makes filtering PRs, issues, etc less efficient. Similarly there isn't a really strong overlap between animation and the other responsibilities of sig-content, whereas the topics proposed for sig-simulation are closely related with a lot of overlap.
- Sig-content is huge (too large?) and sig-core is large enough to support migrating some of its responsibilities without risking it collapsing

This proposal has been discussed at various TSC, sig-core and sig-content meetings. The feedback from the TSC and sig-core has been generally supportive. As I understand it, there has been some concern from sig-content around the overhead of creating a new sig, and finding people to act as chairs. There are several people willing to stand as chairs if the sig is created, so I believe that concern is addressed. I am opening this discussion to allow anyone else to raise a concern if they have one. It will also be discussed again at the next sig-content monthly meeting https://github.com/o3de/sig-content/issues/45.

**Comment by greerdv** [comment]
Draft for how the proposed changes would be reflected in the charters for sig-core, sig-content and sig-simulation
[proposed charter changes.md](https://github.com/o3de/o3de/files/8217276/proposed.charter.changes.md)

I'd be happy to get feedback on this. It's also only kind of stubbed out in terms of scope. The overview and goals need work and any help with those would be much appreciated.

**Reply by lemonade-dm** [reply]
One of the scope bullet points are a task and not an area of responsibility.
Such as
> Consolidate EMotionFX math library with AzCore/math

Wouldn't the scope be to "maintain the math library"?

Also do you have an idea for the Major goal of the sig-simulation?

**Reply by greerdv** [reply]
It was pretty unanimously agreed that AzCore/math should continue to be owned by sig-core, but I wanted to call out here that the EMFX math library would presumably be owned by sig-simulation, since the proposal is for animation to belong there. But I didn't say "maintain EMFX math library" because I think we should be working towards eliminating it once everything is migrate to use AzCore/math.

I have some rough ideas for the goals/overview, but they're quite hard to put into words, which was one of the reasons I'd be happy to get help with those sections. I would point out many of the existing sigs do not have the goals or overview sections of their charters filled out yet (including core, content, graphics-audio, build and platform), so I guess I'm not the only one having difficulty finding the right wording.

**Comment by greerdv** [comment]
Some data on the existing size and scope of sig-core and sig-content https://github.com/o3de/tsc/issues/20#issuecomment-1061913887

**Comment by OBWANDO** [comment]
One approach may be to look at the number of issues and concerns over time. 

Take those relevant areas and consider those as the baseline for creating the new SIG charter, and how we might expect it would balance the load across the SIGs. The only downside is making sure that these are also properly categorized as the "editor" could be an incredibly large bucket all by itself.

**Comment by bhack** [comment]
Check also the recent release of https://github.com/google-research/kubric

**Comment by hultonha** [comment]
Tagging @lsemp3d and @monroegm for visibility as chairs of sig-content and @amzn-pratikpa as chair of sig-core.

We'll be aiming to discuss this in the upcoming sig-content meeting on 22/03/24, the issue has been previously discussed in the sig-core meeting from https://github.com/o3de/sig-core/issues/29
